lib/rubygems/ext/cargo_builder.rb

```diff
@@ -16,10 +16,15 @@ def initialize
     @profile = :release
   end
 
-  def build(extension, dest_path, results, args = [], lib_dir = nil, cargo_dir = Dir.pwd)
+  def build(extension, dest_path, results, args = [], lib_dir = nil, cargo_dir = Dir.pwd,
+    target_rbconfig=Gem.target_rbconfig)
     require "tempfile"
     require "fileutils"
 
+    if target_rbconfig.path
+      warn "--target-rbconfig is not yet supported for Rust extensions. Ignoring"
```

Comment:
what would it take to implement this for rust?

Reply:
Basically, it should be done by replacing `RbConfig` reference with the corresponding value of `target_rbconfig` in `cargo_builder.rb`.  Maybe we need some change in rb-sys side, but wouldn't be big.